blockchain/chaingen/generator.go

```diff
@@ -1020,22 +987,21 @@ func (g *Generator) CalcNextRequiredStakeDifficulty() int64 {
 	// per window and exponentially weight them.
 	var weightedTicketsSum uint64
 	targetTicketsPerWindow := ticketsPerBlock * windowSize
-	block = g.tip
 	for i := int64(0); i < numWindows; i++ {
 		// Since the difficulty for the next block after the current tip
 		// is being calculated and there is no such block yet, the sum
 		// of all new tickets in the first window needs to start with
 		// the number of new tickets in the tip block.
 		var windowNewTickets int64
 		if i == 0 {
-			windowNewTickets = int64(block.Header.FreshStake)
+			windowNewTickets = int64(prevBlock.Header.FreshStake)
```

Comment:
Should be left unchanged and instead bring back the `block := g.tip` above the for loop, except make it `block := prevBlock`.